### ISSUE TITLE
Feature freeze in release procedure

### ISSUE THREAD

**Issue by KennethNielsen** [SoCo/SoCo#581]
Hi all

As evident from #538, there is still room for improvement in our release procedure. I still think it is a good idea to have the regular releases, so as to apply a little pressure on ourselves to have something finished of and to make sure that (even small) contributions make it out the door in a release.

Having a deadline also means that development heated up as the deadline grew nearer. For a spare time project I think that is natural and not necessarily a problem. But merging close to the release also means that those features will not necessarily get tested by the (undoubtedly large) masses of people running bleeding edge SoCo, right of the tip of git master. This is unfortunate, specially since our functional and integrations test are nowhere near complete, so live functional testing is often the only way to catch problems not caught by unit tests or reviews.

**I therefore propose to make a feature freeze (for code, not docs) for the release, 1 or 2 weeks before the planned release date, meaning that after the point is reached, no more new features are merged, only bug fixes (for features added this releases or trivial bug fixes in general).**

**1 or 2 weeks:** Releases tend to be made over a weekend and I think it makes sense to make features freezes after a weekend also, meaning that a Sunday evening is last chance to merge. But should it be 1 or 2 weeks before. Is 5 week days enough time to test and make bug fixes?

Let me know what you think.

**Comment by KennethNielsen** [comment]
@stefankoegl @ghcs27 @amelchio

**Comment by DPH** [comment]
Well done everyone for the effort on this release. 
Sorry I was not able to participate so much this time. 
I think we should allow 2 weeks to test (ie freeze on the Sunday evening before the planned weekend release). 5 days when people have other commitments is a bit tight. If we prove 2 weeks works easily then we could drop it in the future.

Cheers David

**Comment by DPH** [comment]
@stefankoegl - Agree - I support these dates. Gives time to test and tidy release notes.

**Comment by KennethNielsen** [comment]
@stefankoegl yes, feature freeze is a late sunday 29 april as someone wants to work

**Comment by KennethNielsen** [comment]
Seems like we agree. I will leave this open for a little while to give people a possibility to object. After a little while I will put, somewhere in the docs, something about release schedule and this.

@stefankoegl Will you add this date to the release issue and also write that someone should sign of on the functional tests during feature freeze.

**Comment by amelchio** [comment]
With this policy I will be sure to do an integration test of the master branch a week before release, at least when I am waiting for some change to be released. So SoCo gets some testing and I get a chance to validate the release candidate. That seems like a win-win to me. 

...

It would be great, though, with a more steady flow where reviews do not await the deadline. First, it is discouraging to wait around for several months with no comment on your PR. And second, a late review makes it more likely for a contributor to miss the window when some rework is required.

Don't take that as critism of the reviews that _do_ happen. I know this is a small team, hopefully @ghcs27 will soon start merging stuff to speed things up a bit :-)

**Comment by stefankoegl** [comment]
> @stefankoegl Will you add this date to the release issue 

Done

> and also write that someone should sign of on the functional tests during feature freeze.

What procedure do you have in mind for that?
